spec/controllers/user/swaps_controller_spec.rb

```diff
@@ -122,4 +122,61 @@
       end
     end
   end
+
+  context "when users don't have an email" do
```

Comment:
These tests fail, I guess because in this branch users should always have an email address. I'm not quite sure what to do here though, I don't understand enough of how we want this to work! We might need to fix this before merging.